src/packs/checker/pack_checker.rs

```diff
@@ -190,4 +202,34 @@ impl<'a> PackChecker<'a> {
             defining_pack_name: self.defining_pack.unwrap().name.clone(),
         }
     }
+
+    fn interpolate_violation_message(&self, msg: &str) -> String {
```

Comment:
The way this is done in packwerk (ruby) is that the client can configure a custom formatter that takes in a violation and outputs a string. Could we structure it the same way here? e.g. have the violation message formatter be an interface, have a default implementation, then the client can configure their own formatter?

The only thing that's an open question for me is how dependency injection would work in Rust.

Reply:
I'll look into this!